src/pages/Chat/components/rooms/RoomItem.tsx

```diff
@@ -1,21 +1,127 @@
 import type { Room } from "@types";
-import { type FC } from "react";
+import { Fragment, useState, type FC } from "react";
 import Box from "@mui/material/Box";
 import Typography from "@mui/material/Typography";
 import Avatar from "@mui/material/Avatar";
-import AvatarGroup from "@mui/material/AvatarGroup";
+import { generateColor } from "../../utils";
+import Badge from "@mui/material/Badge";
+import Menu from "@mui/material/Menu";
+import MenuItem from "@mui/material/MenuItem";
 
-type Props = { room: Room; onSelect: (id: string) => void };
+import "../../chat.scss";
+
+type Props = {
+  room: Room;
+  onSelect: (id: string) => void;
+  selected: boolean;
+};
+
+export const RoomItem: FC<Props> = ({ room, onSelect, selected }) => {
+  const [contextMenu, setContextMenu] = useState<{
+    mouseX: number;
+    mouseY: number;
+  } | null>(null);
+
+  const handleContextMenu = (e: React.MouseEvent<HTMLDivElement>) => {
+    e.preventDefault();
+    setContextMenu(
+      (e.target as HTMLElement).classList.contains("room-item")
+        ? {
+            mouseX: e.clientX - 2,
+            mouseY: e.clientY - 4,
+          }
+        : null
+    );
+  };
+
+  const handleClose = () => {
+    setContextMenu(null);
+  };
 
-export const RoomItem: FC<Props> = ({ room }) => {
   return (
-    <Box className="room-item">
-      <AvatarGroup spacing={1} total={room.users.length}>
+    <Box
+      className={`room-item ${selected && "selected"}`}
+      onClick={(e) => {
+        if ((e.target as HTMLElement).classList.contains("room-item")) {
+          onSelect(room.id);
+        }
+      }}
+      onContextMenu={handleContextMenu}
+    >
+      <Box
+        sx={{
+          width: "40px",
+          height: "40px",
+          display: "flex",
+          flexFlow: "row wrap",
+          position: "relative",
+          borderRadius: "50%",
+        }}
+      >
         {room.users.map((user) => (
-          <Avatar key={user._id} alt={user.name} src={user.image} />
+          <Fragment key={user._id}>
+            {room.users.length > 1 ? (
+              <>
+                <Box
+                  sx={{
+                    width: 20,
+                    height: 20,
+                    background: user.image
+                      ? `url(${user.image})`
+                      : generateColor(room.id),
+                    backgroundPosition: "center",
+                    backgroundSize: "cover",
+                    color: "#fff",
+                  }}
+                >
+                  {user.image ? "" : user.name[0]}
+                </Box>
+              </>
+            ) : (
+              <Badge
+                overlap="circular"
+                variant="dot"
+                anchorOrigin={{
+                  vertical: "bottom",
```

Comment:
line  like  this  could  be  formatted  into  on e line  anchorOrigin={{ vertical: "bottom",horizontal: "right",}}